### PR TITLE
feat: Add basic config

### LINKED ISSUE #16: Add React and basic libraries
This project was discussed with the team and we decided will be a react PWA but we don't need granular control (for now at least). Also as the primary target stated by Dasher is mobile, we decided to add PWA support.

To start on the shoulder of giants we decided to try [the template available in CRA for PWA](https://create-react-app.dev/docs/making-a-progressive-web-app/) and concentrate on business logic instead.

## Checklist

- [ ] App have `react` at least v17.
- [ ] Add `react-router` with support for hooks
- [ ] Add `emotion` support

## Related discussions

- https://github.com/Azordev/dasher-user/discussions/7
- https://github.com/Azordev/dasher-user/discussions/9

### LINKED ISSUE #29: Add Editor Support, including VSCode
We usually work using different editors, but it looks like one that is used a lot in the team. This is vscode. I want to add support for it and share some sane or interesting defaults.

### LINKED ISSUE #46: Add issues and pull request templates to the project
To ease the creation of issues and pull request, I suggest to add issues and pull request templates in the .github folder

### LINKED ISSUE #47: Add linters to project
We did not decide about what linters to use. I will add all to the checklist and in the comments we could discuss about what the needs.

- [x] Install EsLint
- [ ] Install Pretierr
- [ ] Install CommitLint
- [ ] Install CSpell

### PR DESCRIPTION
**Pull request by denisosuna**
### Description

Please include a summary of the change.

fixes #16 #29 #46 #47

#### Screenshots

If applicable, add screenshots about your changes

#### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] Doesn't break the current code.
- [ ] Passes linters and test, also is building.
- [ ] Doesn't have spelling or grammatical problems.
- [ ] Doesn't have unnecessary comments or debugging code.
- [ ] The branch is updated with the last dev branch changes.

### CODE REVIEW

**Review by Israel-Laguan — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  "license": "Apache-2.0",
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/user-event": "^12.8.3",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-router-dom": "~5.3.0",
+    "react-scripts": "5.0.0",
+    "styled-components": "^5.3.3",
```
Comment (heliomar-pena) [inline package.json]: We have decided to do not use styled components, instead we want to use emotion js

**Review by heliomar-pena — CHANGES_REQUESTED** [review]
I don't see emotion js in package json, can you check if this was added? @denisosuna 

https://emotion.sh/docs/introduction

**Comment by e-azocar** [comment]
@denisosuna about the `spelling`, I think we only should add it to verify the commits, but not verify all the files

**Review by e-azocar — CHANGES_REQUESTED** [review]
@denisosuna  about the spelling, I think we only should add it to verify the commits, but not verify all the files

File: src/App.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ function App() {
           target="_blank"
           rel="noopener noreferrer"
         >
-          Learn React
+          Learn React maro
```
Comment (heliomar-pena) [inline src/App.js]: Maro?

**Comment by heliomar-pena** [comment]
I check and look like the spelling is not working. They have accepted this commit msg: "fix: added kfdsfs things"

imagen

**Comment by e-azocar** [comment]
> I check and look like the spelling is not working. They have accepted this commit msg: "fix: added kfdsfs things"
> 
> imagen

So, the spelling is not working?

**Review by heliomar-pena — APPROVED** [review]
I have tested and is working correctly! Good work!

**Review by e-azocar — APPROVED** [review]

**Pull request merged**